clang/lib/CodeGen/CGCall.cpp

```diff
@@ -6121,6 +6121,8 @@ RValue CodeGenFunction::EmitVAArg(VAArgExpr *VE, Address &VAListAddr,
   VAListAddr = VE->isMicrosoftABI() ? EmitMSVAListRef(VE->getSubExpr())
                                     : EmitVAListRef(VE->getSubExpr());
   QualType Ty = VE->getType();
+  if (Ty->isVariableArrayType())
+    EmitVariablyModifiedType(Ty);
```

Comment:
You can drop the call to EmitVariablyModifiedType in ScalarExprEmitter::VisitVAArgExpr, since it's now redundant.